data/expensePayments.js

```diff
@@ -0,0 +1,146 @@
+const { ObjectId, ObjectID } = require('mongodb');
+const { expensePayment: getExpensePaymentsCollection } = require('../config/mongoCollections');
+const { getExpense } = require('./expenses');
+
+const { QueryError, ValidationError } = require('../utils/errors');
+const { idQuery, parseMongoData, stringifyObjectId } = require('../utils/mongodb');
+const {
+  assertObjectIdString,
+  assertIsValuedString,
+  assertRequiredObject,
+  assertDateString,
+  assertRequiredNumber,
+} = require('../utils/assertion');
+
+const getByObjectId = async (objectId) => {
+  const collection = await getExpensePaymentsCollection();
+  const expensePayment = await collection.findOne(idQuery(objectId));
+  return parseMongoData(expensePayment);
+};
+
+const addExpensePayment = async (data) => {
+  assertRequiredObject(data);
+
+  const { expenseId, amount, currency, notes, method, date } = data;
+  const createdAt = new Date().getTime();
+
+  assertObjectIdString(expenseId, 'Expense id');
+  assertDateString(date, 'Expense payment date');
+  assertRequiredNumber(amount, 'Expense amount');
+  assertIsValuedString(currency, 'Expense curreny');
+  assertIsValuedString(method, 'Payment method');
+
+  const expensePaymentData = {
+    _id: new ObjectId(),
+    expenseId: new ObjectId(expenseId),
+    amount,
+    currency,
+    notes,
+    method,
+    date,
+    createdAt,
+    updatedAt: createdAt,
+  };
+
+  const expense = await getExpense(expenseId);
+  if (!expense) {
+    throw new QueryError(`Expense not exist for expense ID(${expenseId})`);
+  }
+
+  const collection = await getExpensePaymentsCollection();
+
+  const payment = await collection.find({ expenseId: new ObjectId(expenseId) }).toArray();
```

Comment:
Isn't a payment in a one-to-one relationship to an expense?

Reply:
It is just to check that whether expense exist or not